diesel/src/sqlite/connection/mod.rs

```diff
@@ -406,6 +407,41 @@ impl SqliteConnection {
             .register_collation_function(collation_name, collation)
     }
 
+    /// Deserialize an SQLite database from a byte buffer.
+    ///
+    /// This function takes a byte slice and attempts to deserialize it into a SQLite database.
```

Comment:
The documentation needs to mention that it will open the database in read only mode. It might be even a good idea to change the method name to `deserialize_read_only_database_from_buffer()`